From 19fa92b4e5bf88c8ddf483d4b95897717af5410f Mon Sep 17 00:00:00 2001
From: Vekhir <134215107+Vekhir@users.noreply.github.com>
Date: Wed, 18 Oct 2023 08:15:27 +0200
Subject: [PATCH] Update supported versions (#10)

- Add support for GHC 9.8. 9.8 uses Cabal 3.10.2.0 which is a strict bug fix release, so no changes expected.
- Fully support Cabal 3.6. The previous change was actually introduced in 3.8, so all 3.6 releases have the same behaviour.
---
 app/Main.hs              |  2 +-
 uusi.cabal               | 10 +++++-----
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/app/Main.hs b/app/Main.hs
index 7de3e3a..72db961 100644
--- a/app/Main.hs
+++ b/app/Main.hs
@@ -8,7 +8,7 @@ module Main (main) where
 import Control.Monad (unless, when)
 import qualified Data.Text as T
 import qualified Data.Text.IO as T
-#if !MIN_VERSION_Cabal(3,6,0)
+#if !MIN_VERSION_Cabal(3,8,0)
 import Distribution.PackageDescription.Parsec (readGenericPackageDescription)
 #else
 import Distribution.Simple.PackageDescription (readGenericPackageDescription)
diff --git a/uusi.cabal b/uusi.cabal
index 40a81ea..4285fae 100644
--- a/uusi.cabal
+++ b/uusi.cabal
@@ -21,11 +21,11 @@ extra-doc-files:
   README.md
 
 tested-with:
-  GHC ==8.6.5
-   || ==8.8.3
-   || ==8.10.5
-   || ==9.0.1
+  GHC ==9.0.2
+   || ==9.2.8
+   || ==9.4.7
    || ==9.6.2
+   || ==9.8.1
 
 source-repository head
   type:     git
@@ -34,7 +34,7 @@ source-repository head
 common common-options
   build-depends:
     , base   >=4.8 && <5
-    , Cabal  ^>=3.2 || ^>=3.4 || ==3.6.3 || ^>= 3.8 || ^>= 3.10
+    , Cabal  ^>=3.2 || ^>=3.4 || ^>=3.6 || ^>= 3.8 || ^>= 3.10
     , text
 
   ghc-options:
